compiler/src/iree/compiler/Codegen/SPIRV/test/lowering_matmul_promotion.mlir

```diff
@@ -51,34 +51,49 @@ hal.executable @matmul_f32_128x256x64 {
   }
 }
 
-//       CHECK-DAG: #[[MAP:.+]] = affine_map<(d0) -> ((d0 floordiv 16) mod 2)>
-//       CHECK-DAG: #[[TRANSLATION:.+]] = #iree_codegen.translation_info<SPIRVMatmulPromoteVectorize pipeline_depth = 2>
+//       CHECK-DAG: #[[MAP:.+]] = affine_map<(d0) -> ((d0 floordiv 16) mod 3)>
+//       CHECK-DAG: #[[TRANSLATION:.+]] = #iree_codegen.translation_info<SPIRVMatmulPromoteVectorize pipeline_depth = 2 store_stage = 0>
 //           CHECK: hal.executable.export public @matmul_f32_128x256x64
 //      CHECK-SAME:   translation_info = #[[TRANSLATION]]
 //      CHECK-SAME:   workgroup_size = [16 : index, 8 : index, 1 : index]
 //           CHECK: func.func @matmul_f32_128x256x64()
 //           CHECK:   %[[CST0:.+]] = arith.constant 0.000000e+00 : f32
-//           CHECK:   memref.alloc() : memref<2x64x20xf32, 3>
-//           CHECK:   memref.alloc() : memref<2x16x68xf32, 3>
+//           CHECK:   memref.alloc() : memref<3x64x20xf32, 3>
+//           CHECK:   memref.alloc() : memref<3x16x68xf32, 3>
+//   CHECK-COUNT-8:   vector.transfer_write %{{.+}}, %{{.+}} {in_bounds = [true]} : vector<4xf32>, memref<128x256xf32>
+//   CHECK-COUNT-8:   vector.transfer_read %{{.+}}, %[[CST0]] {in_bounds = [true]} : memref<128x256xf32>, vector<4xf32>
+//           CHECK:   vector.transfer_read %{{.+}}, %[[CST0]] {__pipelining_first_stage__, in_bounds = [true]} : memref<128x512xf32>, vector<4xf32>
+//           CHECK:   vector.transfer_write %{{.+}}, %{{.+}} {__pipelining_first_stage__, in_bounds = [true]} : vector<4xf32>, memref<3x64x20xf32, 3>
+//           CHECK:   vector.transfer_read %{{.+}}, %[[CST0]] {__pipelining_first_stage__, in_bounds = [true]} : memref<512x256xf32>, vector<4xf32>
+//           CHECK:   vector.transfer_write %{{.+}}, %{{.+}} {__pipelining_first_stage__, in_bounds = [true]} : vector<4xf32>, memref<3x16x68xf32, 3>
+//           CHECK:   gpu.barrier {__pipelining_first_stage__}
```

Comment:
IIUC, we should be writing to different slices of the workgroup memory before and after this barrier. So is this barrier necessary here? @ThomasRaoux

Reply:
yeah they're writing to different slices, but I thought I removed this barrier before pipelining so I'm not yet sure why it ends up here.